lib/SILOptimizer/Mandatory/SILGenCleanup.cpp

```diff
@@ -0,0 +1,78 @@
+//===--- SILGenCleanup.cpp ------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2019 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+// Perform peephole-style "cleanup" to aid SIL diagnostic passes.
+//
+//===----------------------------------------------------------------------===//
+
+#define DEBUG_TYPE "silgen-cleanup"
+
+#include "swift/SIL/SILInstruction.h"
+#include "swift/SILOptimizer/PassManager/Transforms.h"
+#include "swift/SILOptimizer/Utils/CanonicalizeInstruction.h"
+
+using namespace swift;
+
+// Define a CanonicalizeInstruction subclass for use in SILGenCleanup.
+struct SILGenCanonicalize : CanonicalizeInstruction {
```

Comment:
Can you make this a final inheritance and mark all of the overridden methods with override?